app/controllers/home_controller.rb

```diff
@@ -1,5 +1,13 @@
 class HomeController < ApplicationController
-  def index; end
+  include ApplicationHelper
+  before_action :require_user!, only: :app
+
+  def index
+    return redirect_to app_path if current_user.present?
+    return redirect_to auth.sign_in_path, flash: { error: 'Expired!' } if session_expired?
```

Comment:
Let's make the flash message more informative: "The last login link you tried using expired. Please request a new one."

Reply:
@rogergraves I don't show this flash for users and use it only for condition in new.html.erb
`<% if flash.to_h["error"] == 'Expired!' %>
      <h3 class="muted my-4">Your login link may have expired. Click here to get a new one</h3>
    <% else %>`